### PR TITLE
Add 19031

### PR DESCRIPTION
**Pull request by troygiorshev**

### CODE REVIEW

**Comment by jonatack** [comment]
Great, @troygiorshev -- will review this tomorrow early, e.g. in ~10 hours, if it's still open.

File: _posts/2020-07-15-#19031.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+host: troygiorshev
+status: upcoming
+commit:
+---
+
+## Notes
+
+  * [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is the implementation of [BIP 155](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki), the `addrv2` message.
+  * The smaller PR that's currently being worked on, and should be reviewed first, is [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360).
+  * Ultimately this will allow us to work with [Tor v3](https://github.com/bitcoin/bitcoin/issues/18884), along with many other different types of networks.
```
Comment (jonatack) [inline _posts/2020-07-15-#19031.md]: If you have time, don't hesitate to expand this section by providing more resources in the notes: more background, context, info on the topic. See sessions like:

- https://bitcoincore.reviews/15681
- https://bitcoincore.reviews/16702
- https://bitcoincore.reviews/17639

Comment (jnewbery) [reply _posts/2020-07-15-#19031.md]: I agree. A few words about what addr v2 and tor v3 are would be helpful.

**Comment by troygiorshev** [comment]
Thanks for the suggestions everyone.  I've rewrote and expanded the notes, and added a "suggested approach" for reviewing.

File: _posts/2020-07-15-#19031.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  * Read [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki).  There is ongoing discussion on [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) happening at [BIP PR 907](https://github.com/bitcoin/bips/pull/907).  Because of this, please review [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) at any of the links in these notes as they reflect the current status of the BIP.
+  * See [this comment](https://github.com/bitcoin/bips/pull/907#issuecomment-611924492) in BIP PR 907 about how to signal support for `addrv2`.  The final decision can be seen [here](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki#Signaling_support_and_compatibility)
+  * Read through all of the conceptual comments in 19031
+  * Skim commits 2-4 of PR 19031.  The goal should be to get context for what future PRs will cover, and to see how the implementation of [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki) matches the specification.
```
Comment (jonatack) [inline _posts/2020-07-15-#19031.md:26]: For each of the items in this list, maybe use a full stop consistently or not at all.

Comment (jonatack) [inline _posts/2020-07-15-#19031.md:26]: s/cover, and/cover and/

File: _posts/2020-07-15-#19031.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+   You're always encouraged to put your PR review on GitHub, even after it has
+   been merged.
+
+2. What are the differences between `addr` and the new `addrv2`?
+
+3. Do you agree with how [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is being split up into smaller PRs?
```
Comment (jonatack) [inline _posts/2020-07-15-#19031.md:37]: Perhaps be more explicit in this question.

**Comment by jonatack** [comment]
Pinging @vasild in case he'd like to have a look.

**Comment by jnewbery** [comment]
Thanks for the thorough review @jonatack ! I'm not sure if @troygiorshev is going to be online this weekend to respond. If there's no update on here by my evening I'll go ahead and merge this, and we can apply fixups on Monday.

In the meantime I've pushed the template page to master, so at least participants can see which PR we're covering.

**Comment by jonatack** [comment]
SGTM! Have it up and tweeted today for sure.

File: _posts/2020-07-15-#19031.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+commit:
+---
+
+## Notes
+
+* [PR 19031](https://github.com/bitcoin/bitcoin/pull/19031) is the implementation of [BIP 155](https://github.com/bitcoin/bips/blob/9286b5254317d9e73fb25c5f0acd2b2d9937843e/bip-0155.mediawiki), the `addrv2` message.  PR 19031 is being used as a roadmap, where [PR 19360](https://github.com/bitcoin/bitcoin/pull/19360) is currently ready for review.
+
+* The `addr` message is currently used to relay IP addresses across the network.  It uses only IPv6 addresses, where IPv4 addresses are given as IPv4-mapped IPv6 addresses.  More information about it can be found [here](https://btcinformation.org/en/developer-reference#addr).  The `addrv2` message will allow for longer variable length addresses.
+
+* Bitcoin currently has Tor integration.  Relevant to this PR is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address.  Ultimately these changes will allow us to work with [Tor v3](https://github.com/bitcoin/bitcoin/issues/18884), along with many other different types of networks.
```
Comment (jonatack) [inline _posts/2020-07-15-#19031.md:19]: suggest: work with Tor v3 ([meta issue here](https://github.com/bitcoin/bitcoin/issues/18884))


I think the notes ought to describe Tor v3 or link to a good resource. One resource I found was: https://trac.torproject.org/projects/tor/wiki/doc/NextGenOnions; perhaps you know a better one.

**Comment by jonatack** [comment]
If no one is opposed, I propose to rebase this now and merge it, and add a commit with the edits. Then send out the tweet.

**Comment by jonatack** [comment]
Ha!

**Comment by jnewbery** [comment]
Merged. @troygiorshev - do you mind looking at the feedback tomorrow and opening a PR if you agree with the proposed changes?

**Pull request merged**